search/src/Components/forms/AgContextForm.js

```diff
@@ -140,10 +140,17 @@ class AgContextForm extends Component {
               renderers={renderers}
               cells={materialCells}
               onChange={e => {
+                  if (this.props.handleCocoFormValidation){
+                    if (e.errors.length === 0) {
+                      this.props.handleCocoFormValidation('agcontexts', true);
+                    } else {
+                      this.props.handleCocoFormValidation('agcontexts', false);
+                    }
+                  }
```

Comment:
This can be shortened to:
```suggestion
                  if (this.props.handleCocoFormValidation){
                    this.props.handleCocoFormValidation('agcontexts', e.errors.length == 0);
                  }
```

I think `handleCocoFormValidation` could have a simpler name, e.g. `handleValidation`. And I'm surprised we need to pass back the name of the form, when surely the containing component has this context.

But I wonder why we don't just use the existing onChange handler to report back?

Reply:
Cause the only binding is upload_type + activeStep, and retrieve the actual form name from the getSteps() but we have separate validation checking for uploaders so this can be the best and quickest solution for now.
what do you mean by report back? log?

Reply:
I think that the logic here is quite convoluted. I guess we will clean a lot of it up if we eventually support more upload formats.